### PR TITLE
OpenAMP resource table sample fixes

### PR DESCRIPTION
**Pull request by glneo**
Some hopefully minor fixups to this sample application. Done in preparation of adding a new sample application that is based on this but uses the new Mailbox framework instead of IPM.

### CODE REVIEW

File: samples/subsys/ipc/openamp_rsc_table/src/main_remote.c [hunk 1]
```diff
@@ -156,19 +156,19 @@ int platform_init(void)
 
 	status = metal_init(&metal_params);
 	if (status) {
-		LOG_DBG("metal_init: failed: %d\n", status);
+		LOG_WRN("metal_init: failed: %d", status);
```
Comment (arnopo) [inline samples/subsys/ipc/openamp_rsc_table/src/main_remote.c:149]: use LOG_ERR instead as we report an error, same for other LOG_WRN

File: samples/subsys/ipc/openamp_rsc_table/src/main_remote.c [hunk 2]
```diff
@@ -53,26 +54,17 @@ static const struct device *const ipm_handle =
...
-	.regions = {
-		{.virt = NULL}, /* shared memory */
-		{.virt = NULL}, /* rsc_table memory */
-	},
-	.node = { NULL },
-	.irq_num = 0,
-	.irq_info = NULL
-};
+static struct metal_io_region shm_io_data; /* shared memory */
+static struct metal_io_region rsc_io_data; /* rsc_table memory */
```
Comment (arnopo) [inline samples/subsys/ipc/openamp_rsc_table/src/main_remote.c:58]: what about naming them `shm_io` and `rsc_io ` and remove ` static struct metal_io_region *shm_io ` and `static struct metal_io_region *shm_io`?

**Review by arnopo — DISMISSED** [review]
Very good optimization! I tested it with success.
2 minor comments, with that LGTM

File: samples/subsys/ipc/openamp_rsc_table/src/main_remote.c (hunk 2 above)
Comment (glneo) [reply samples/subsys/ipc/openamp_rsc_table/src/main_remote.c:58]: I thought about that originally, but this causes less churn at each point they are used (removing all the dereferences). And at some point in the future I want to make `metal_io_region` an opaque struct. So these would need to become a pointer again, so less churn changing them back too.

**Review by carlocaione — APPROVED** [review]

**Review by arnopo — COMMENTED** [review]
@glneo : please have look to the CI test that fails. Seems that you have to add your copyright to be compliant.

**Review by arnopo — APPROVED** [review]

**Pull request merged**